store/tikv/region_request.go

```diff
@@ -194,7 +194,7 @@ func (s *RegionRequestSender) onRegionError(bo *Backoffer, ctx *RPCContext, regi
 		if notLeader.GetLeader() != nil {
 			boType = BoUpdateLeader
 		} else {
-			boType = BoRegionMiss
+			boType = boRegionStale
```

Comment:
Define a new type for it, like `boNoLeader`.

Reply:
Seems we should check if `notLeader.GetLeader()` is `nil` before updating the cache.
/cc @disksing